### PR TITLE
[ES-7538] Add Jenkinsfile

### PR DESCRIPTION
**Pull request by Sisyphus25**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  environment {
+    SEARCH_PLACEMENT_ID = 2967
+    SEARCH_IM_URL = "https://cdn.visenze.com/images/widget-2.jpg"
+    REC_PLACEMENT_ID = 1823
+    REC_PID = "184827-09"
+    ENDPOINT = "https://search-dev.visenze.com"
+  }
+
+  tools {
+    nodejs('NodeJS14')
```
Comment (wkurniawan07) [inline Jenkinsfile]: Should be `NodeJS16`

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            })
+          }
+        }
+      }
+    }
+
+    stage('Tag') {
+      steps {
+        script {
+          if (env.BRANCH_NAME == 'master') {
```
Comment (wkurniawan07) [inline Jenkinsfile]: There's quite some unusual practice in this repo:
- Both `master` and `develop` co-existing.
- `master` as the release-able state, and yet at the same time the default branch.
  - If `master` acts like the `production` branch in many of our other repos, it shouldn't be the default branch.
  - If `master` is the default branch and we release on demand e.g. if version number is bumped, which is quite a common practice in open source projects, then there needs to be more check here than just the branch name.

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 2 above)
Comment (Sisyphus25) [reply Jenkinsfile]: Master is the old default branch, I will update it.

Comment (wkurniawan07) [reply Jenkinsfile]: @Sisyphus25 with that being done, maybe we can use the opportunity to rename the branch? `production` or `stable` sounds better.

Comment (Sisyphus25) [reply Jenkinsfile]: It's changed now.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            })
+          }
+        }
+      }
+    }
+
+    stage('Tag') {
+      steps {
+        script {
+          if (env.BRANCH_NAME == 'production') {
```
Comment (wkurniawan07) [inline Jenkinsfile]: Instead of using if block here, you can consider using `when` block for the stage itself

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            def version = getVersion()
+            build(
+              job: 'devops_github_utility_create_release',
+              parameters: [
+                string(name: 'GITHUB_REPO', value: GIT_REPO),
+                string(name: 'TAG_NAME', value: version),
+                string(name: 'TARGET_COMMITISH', value: 'master'),
```
Comment (wkurniawan07) [inline Jenkinsfile]: Should be `production`, or better yet, just use `env.BRANCH_NAME`

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]

**Review by wkurniawan07 — APPROVED** [review]
LGTM

**Pull request merged**